packages/format-library/src/link/test/index.native.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * External dependencies
+ */
+import { Keyboard, Platform } from 'react-native';
+import { render, fireEvent, waitFor } from 'test/helpers';
+
+/**
+ * WordPress dependencies
+ */
+import { Slot, SlotFillProvider } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { link } from '../index';
+
+const { edit: LinkEdit } = link;
+
+// Simplified tree to render link format component
+const LinkEditSlot = ( props ) => (
+	<SlotFillProvider>
+		<Slot name="RichText.ToolbarControls.link" />
+		<LinkEdit { ...props } />
+	</SlotFillProvider>
+);
+
+beforeAll( () => {
+	jest.useFakeTimers();
+	jest.spyOn( Keyboard, 'dismiss' );
+} );
+
+describe( 'Android', () => {
+	it( 'ensures smooth back animation', async () => {
+		const screen = render(
+			<LinkEditSlot
+				activeAttributes={ {} }
+				onChange={ () => {} }
+				value={ {
+					text: '',
+					formats: [],
+					replacements: [],
+				} }
+			/>
+		);
+		fireEvent.press( screen.getByA11yLabel( 'Link' ) );
+		fireEvent.press(
+			screen.getByA11yLabel( 'Link to, Search or type URL' )
+		);
+		// Await back button to allow async state updates to complete
+		const backButton = await waitFor( () =>
+			screen.getByA11yLabel( 'Go back' )
+		);
+		Keyboard.dismiss.mockClear();
+		fireEvent.press( backButton );
+
+		expect( Keyboard.dismiss ).toHaveBeenCalledTimes( 1 );
```

Comment:
@dcalhoun , wdyt about adding a line comment here to explain how the single firing of the keyboard dismissal means that `ensures smooth back animation` has been met? I mean, perhaps it makes sense if one goes through the code but, it might not be easy to deduce if the code paths change or the runtime flow is altered for some reason.

Reply:
I took the approach of expanding the test description in https://github.com/WordPress/gutenberg/pull/36328/commits/0800158a5998faf2f5d747714d9dea28458c6c50. Let me know if you believe this could still use a more descriptive title and/or a comment above the assertion.

Reply:
Aha, works for me, nice 👍. Thanks!